### PR TITLE
Rafi devel

### PR DESCRIPTION
**Pull request by radicalrafi**
Implemented cumsum and ceil operations I would like a review of the code before going forward with the next ones .

### CODE REVIEW

**Comment by TheOriginalAlex** [comment]
Please merge it against the `develop` branch.

File: syft/tensor.py [hunk 1]
```diff
@@ -1,85 +1,169 @@
...
+
+    def sum(self, dim=None):
+        """Returns the sum of all elements in the input array."""
+        if self.encrypted:
+            return NotImplemented
+
+        if dim is None:
+            return self.data.sum()
+        else:
+            return self.data.sum(axis=dim)
```
Comment (erfannoury) [inline syft/tensor.py]: I think there is no need for this `if` and `else`. Axis argument to `numpy.sum` can be `None`.

File: syft/tensor.py [hunk 2]
```diff
@@ -1,85 +1,169 @@
...
+
+    def ceil(self):
+        """Returns the ceilling of the input tensor elementwise."""
+        if self.encrypted:
+            return NotImplemented
+        return np.ceil(self.data)
```
Comment (erfannoury) [inline syft/tensor.py]: Also add a new line here.

**Comment by radicalrafi** [comment]
I fixed the couple errors that @erfannoury has spotted :+1:  and fixed a conflict I also changed the merge base to develop branch . sorry about that

File: syft/math.py [hunk 3]
```diff
@@ -45,3 +45,41 @@ def matmul(tensor1, tensor2):
...
+    """
+    Returns the ceilling input tensor,element wise .
+
+    Ceilling of an input scalar is the smallest integer such as :
+    for each floating pount number x : a >= x
+
+    Behavior is independent of a tensor's shape.  
+
+    :input: TensorBase tensor\n
+    :return: TensorBase tensor
```
Comment (erfannoury) [inline syft/math.py]: Sorry I'm somehow nitpicking :D
Put a space after `,` (lines 52, 69, 85 in this file, lines 23, 25, 30, 31 in `test_math.py`) and remove space before period and colon (lines 52, 54, 64 in this file).
Also try to make the code `pep8`-compliant.

**Comment by erfannoury** [comment]
@radicalrafi thank you for the changes. I have also added a small comment that would be good to address, although not critical. 
Oh, one more problem. You have mistakenly also added this file to the commit: `__pycache__/__init__.cpython-36.pyc`. Usually it should've been ignored, but haven't yet checked the `.gitignore` file.

**Comment by erfannoury** [comment]
@iamtrask I think this PR needs further action, especially removal of the `*.pyc` file.

**Pull request merged**